src/pointer-analysis/value_set_fi.cpp

```diff
@@ -418,26 +419,27 @@ void value_set_fit::get_value_set_rec(
   }
   else if(expr.id()==ID_index)
   {
-    assert(expr.operands().size()==2);
-
-    const typet &type = expr.op0().type();
+    const typet &type = to_index_expr(expr).array().type();
 
     DATA_INVARIANT(type.id()==ID_array ||
                    type.id()=="#REF#",
                    "operand 0 of index expression must be an array");
 
-    get_value_set_rec(expr.op0(), dest, "[]"+suffix,
-                      original_type, ns, recursion_set);
+    get_value_set_rec(
+      to_index_expr(expr).array(),
+      dest,
+      "[]" + suffix,
+      original_type,
+      ns,
+      recursion_set);
 
     return;
   }
   else if(expr.id()==ID_member)
   {
-    assert(expr.operands().size()==1);
-
-    if(expr.op0().is_not_nil())
+    if(to_member_expr(expr).compound().is_not_nil())
```

Comment:
It would be slightly better to make variable which is a reference to `to_member_expr(expr).compound()`

Reply:
This hasn't been done